app/sessions.ts

```diff
@@ -0,0 +1,33 @@
+// app/sessions.ts
+import { createCookieSessionStorage } from "@remix-run/node"; // or cloudflare/deno
+
+type SessionData = {
+  userId: string;
+};
+
+type SessionFlashData = {
+  error: string;
+};
+
+const { getSession, commitSession, destroySession } =
+  createCookieSessionStorage<SessionData, SessionFlashData>({
+    // a Cookie from `createCookie` or the CookieOptions to create one
+    cookie: {
+      name: "__session",
+
+      // all of these are optional
+      domain: "remix.run",
+      // Expires can also be set (although maxAge overrides it when used in combination).
+      // Note that this method is NOT recommended as `new Date` creates only one date on each server deployment, not a dynamic date in the future!
+      //
+      // expires: new Date(Date.now() + 60_000),
+      httpOnly: true,
+      maxAge: 60,
+      path: "/",
+      sameSite: "lax",
+      secrets: ["s3cret1"],
+      secure: true,
```

Comment:
is this right? I suspect most of the time we will not have a secure connection since it will be local-only network access.

Reply:
interesting, this wasn't a problem during dev, looks like the clients (browsers) treat localhost as a special case:

https://stackoverflow.com/questions/65006250/secure-cookie-on-http-localhost

I think this'll be ok for now. If this is ever hosted as web app it's unlikely it will be without SSL